File: Data Structures & Algorithms/products-of-array-discluding-self/submission-2.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        vector<int> res;
        int product = 1;
        int zerocount = 0;
        for(int i = 0; i < nums.size(); i++){
            if(nums[i] == 0){
                zerocount++;
            } else{
                product *= nums[i];
            }
        }
        for(int i = 0; i < nums.size(); i++){
            if(zerocount > 1){
                res.push_back(0);
            }
            else if(zerocount == 1){
                if(nums[i] == 0){
                    res.push_back(product);
                }else{
                    res.push_back(0);
                }
            }else{
                res.push_back(product / nums[i]);
            }
        }
        return res;
    }
};

File: Data Structures & Algorithms/sort-colors/submission-2.java
class Solution {
    public void sortColors(int[] nums) {
        int l = 0;
        int m = 0;
        int h = nums.length - 1;
        while(m <= h){
            if(nums[m] == 0){
                int t = nums[l];
                nums[l] = nums[m];
                nums[m] = t;
                l++;
                m++;
            }
            else if(nums[m] == 2){
                int t = nums[h];
                nums[h] = nums[m];
                nums[m] = t;
                h--;
            }
            else {
                m++;
            }

        }
    }
}